helpers/generate.py

```diff
@@ -276,7 +276,21 @@ def generate(args, root, frame = 0, return_latent=False, return_sample=False, re
                         results.append(c.clone())
 
                     for x_sample in x_samples:
-                        x_sample = 255. * rearrange(x_sample.cpu().numpy(), 'c h w -> h w c')
-                        image = Image.fromarray(x_sample.astype(np.uint8))
+                        def uint_number(datum, number):
+                            if number == 8:
+                                datum = Image.fromarray(datum.astype(np.uint8))
+                            elif number == 32:
+                                datum = datum.astype(np.float32)
```

Comment:
why is `Image.fromarray()` not used on this line and also `L:285`?

We can also simplify this checking of numbers by creating a dictionary, and using that to refer to the correct `np.uintN` type:

```py
np_uint_map = {
    8: np.uint8,
    16: np.uint16,
    32: np.uint32
}

datum = Image.fromarray(
    datum.astype(np_uint_map[number])
)

```

Reply:
Image.fromarray() creates a PIL image.

PIL image is only used for 8bpc.